django/views/decorators/http.py

```diff
@@ -112,7 +129,25 @@ def inner(request, *args, **kwargs):
                 if res_etag:
                     response.headers.setdefault("ETag", res_etag)
 
-            return response
+        if iscoroutinefunction(func):
+
+            @wraps(func)
+            async def inner(request, *args, **kwargs):
+                response, res_etag, res_last_modified = _pre_process_request(request)
```

Comment:
We should pass `*args` and `**kwargs` to keep the current logic.
```suggestion
                response, res_etag, res_last_modified = _pre_process_request(request, *args, **kwargs)
```